content/influxdb/cloud-dedicated/reference/internals/storage-engine.md

```diff
@@ -0,0 +1,178 @@
+related:
+  - /influxdb/cloud-dedicated/admin/custom-partitions/
+---
+
+The InfluxDB v3 storage engine is a real-time, columnar database optimized for
+time series data built in [Rust](https://www.rust-lang.org/) on top of
+[Apache Arrow](https://arrow.apache.org/) and
+[DataFusion](https://arrow.apache.org/datafusion/user-guide/introduction.html).
+It supports infinite tag cardinality (number of unique tag values), real-time
+queries, and is optimized to reduce storage cost.
+
+- [Storage engine diagram](#storage-engine-diagram)
+- [Storage engine components](#storage-engine-components)
+  - [Ingester](#ingester)
+  - [Querier](#querier)
+  - [Catalog](#catalog)
+  - [Object store](#object-store)
+  - [Compactor](#compactor)
+- [Scaling strategies](#scaling-strategies)
+  - [Vertical scaling](#vertical-scaling)
+  - [Horizontal scaling](#horizontal-scaling)
+
+## Storage engine diagram
+
+{{< svg "/static/svgs/v3-storage-architecture.svg" >}}
+
+## Storage engine components
+
+- [Ingester](#ingester)
+- [Querier](#querier)
+- [Catalog](#catalog)
+- [Object store](#object-store)
+- [Compactor](#compactor)
+
+### Ingester
+
+The Ingester processes line protocol submitted in write requests and persists
+time series data to the [Object store](#object-store).
+In this process, the Ingester does the following:
+
+- Queries the [Catalog](#catalog) to identify where data should be persisted and
+  to ensure the schema of the line protocol is compatible with the [schema](/influxdb/cloud-dedicated/reference/glossary/#schema) of
+  persisted data.
+- Accepts or [rejects](/influxdb/cloud-dedicated/write-data/troubleshoot/#troubleshoot-rejected-points) points in the write request and generates a [response](/influxdb/cloud-dedicated/write-data/troubleshoot/).
+- Processes line protocol and persists time series data to the
+  [Object store](#object-store) in Apache Parquet format. Each Parquet file
+  represents a _partition_--a logical grouping of data.
+- Makes [yet-to-be-persisted](/influxdb/cloud-dedicated/reference/internals/durability/#data-ingest) data available to [Queriers](#querier) to ensure
+  leading edge data is included in query results.
+- Maintains a short-term [write-ahead log (WAL)](/influxdb/cloud-dedicated/reference/internals/durability/) to prevent data loss in case of a
+  service interruption.
+
+##### Ingester scaling strategies
+
+The Ingester can be scaled both [vertically](#vertical-scaling) and
+[horizontally](#horizontal-scaling).
+Horizontal scaling increases write throughput and is typically the most
+effective scaling strategy for the Ingester.
+
+### Querier
+
+The Querier handles query requests and returns query results for requests.
+It supports both SQL and InfluxQL through
+[Apache Arrow DataFusion](https://arrow.apache.org/datafusion/user-guide/introduction.html).
+
+#### Query life cycle
+
+At query time, the querier:
+
+1.  Receives the query request and builds a query plan.
+2.  Queries the [Ingester](#ingester) to:
```

Comment:
I wonder if you want to make `Ingester` her plural because the Querier (the one that handle the query) talks to all ingesters in the cluster for not-yet-persisted data